sdk/search/Azure.Search.Documents/src/Models/SearchField.cs

```diff
@@ -1,12 +1,137 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+using System.Collections.Generic;
 using Azure.Core;
 
 namespace Azure.Search.Documents.Models
 {
+    /// <summary>
+    /// Represents a field in an index definition, which describes the name, data type, and search behavior of a field.
+    /// <para>
+    /// When creating an index, instead use the <see cref="SimpleField"/> and <see cref="ComplexField"/> classes to help you more easily create a <see cref="SearchIndex"/>.
+    /// </para>
+    /// </summary>
     [CodeGenSchema("Field")]
     public partial class SearchField
     {
+        // TODO: Replace constructor and read-only properties when https://github.com/Azure/autorest.csharp/issues/554 is fixed.
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SearchField"/> class.
+        /// </summary>
+        /// <param name="name">The name of the field, which must be unique within the index or parent field.</param>
+        /// <param name="type">The data type of the field.</param>
+        /// <exception cref="ArgumentException"><paramref name="name"/> is an empty string.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/> is null.</exception>
+        public SearchField(string name, DataType type)
+        {
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+
+            Name = name;
+            Type = type;
+        }
+
+        // TODO: Remove when https://github.com/Azure/autorest.csharp/issues/582 is fixed.
+        private SearchField()
+        {
+        }
+
+        /// <summary>
+        /// Gets the name of the field.
+        /// </summary>
+        [CodeGenSchemaMember("name")]
+        public string Name { get; }
+
+        /// <summary>
+        /// Ge the data type of the field.
+        /// </summary>
+        [CodeGenSchemaMember("type")]
+        public DataType Type { get; }
```

Comment:
Should we call the property name `DataType` as well to help add the sense of separation from .NET's types?

Reply:
That's atypical. I always see `Type` as a property, unless it needs disambiguation with similar properties on the same type, and there's no conflict with `Type GetType()`. How bad do you want this?

Reply:
I don't feel really strongly about it - I just wonder if the name `DataType` might be a clearer hint given that we'll also have a `FieldBuilder` interface that uses the CLR types.

Reply:
If we're going to rename this or `DataType`, what about the type is `FieldDataType` (better than sticking "Search" in front of everything, and relates it to the various "*Field" types better) and this member as "DataType" since it's the type of the data in the field?

Reply:
We'll take this offline in APIView, though.